File: src/main/java/lzhang/leetcode/Twenty.java
package lzhang.leetcode;

import java.util.Stack;

public class Twenty {
  public boolean isValid(String s) {
    Stack<Character> stack=new Stack<>();
    for (int i=0; i<s.length(); i++){
      switch(s.charAt(i)){
      case '(':
        stack.add('(');
        break;
      case '[':
        stack.add('[');
        break;
      case '{':
        stack.add('{');
        break;
      case ')':
        if (stack.isEmpty() || stack.pop()!='('){
          return false;
        }
        break;
      case ']':
        if (stack.isEmpty() || stack.pop()!='['){
          return false;
        }
        break;
      case '}':
        if (stack.isEmpty() || stack.pop()!='{'){
          return false;
        }
        break;
        default:
          break;
      }
    }
    return stack.isEmpty();
  }
}

File: javascript/twenty.js
/**
 * @param {string} s
 * @return {boolean}
 */

var isValid = function(s) {
	var stack=[];
	for (var i=0; i<s.length; i++){
		switch(s[i]){
			case '(':
				stack.push('(');
				break;
			case '[':
				stack.push('[');
				break;
			case '{':
				stack.push('{');
				break;
			case ')':
				if (stack[stack.length-1]!=='('){
					return false;
				}
				stack.pop();
				break;
			case ']':
				if (stack[stack.length-1]!=='['){
					return false;
				}
				stack.pop();
				break;
			case '}':
				if (stack[stack.length-1]!=='{'){
					return false;
				}
				stack.pop();
				break;
		}
	}
	return !stack.length;
};
